airflow/example_dags/example_xcom.py

```diff
@@ -63,12 +61,21 @@ def puller(**kwargs):
 
 
 push1 = PythonOperator(
-    task_id='push', dag=dag, python_callable=push)
+    task_id='push',
+    dag=dag,
+    python_callable=push,
+)
 
 push2 = PythonOperator(
-    task_id='push_by_returning', dag=dag, python_callable=push_by_returning)
+    task_id='push_by_returning',
+    dag=dag,
+    python_callable=push_by_returning,
+)
 
 pull = PythonOperator(
-    task_id='puller', dag=dag, python_callable=puller)
+    task_id='puller',
+    dag=dag,
+    python_callable=puller,
+)
 
-pull.set_upstream([push1, push2])
+pull << [push1, push2]
```

Comment:
Yes, this operation is implemented by [`__lshift__()`](https://github.com/apache/incubator-airflow/blob/39a9365b6250e588363e59dd6e332e02a28f411d/airflow/models.py#L2658-L2671) which calls `set_upstream()`.

Reply:
Personally I don't think `<<` is better than `.set_upstream`. It's less explicit and introduces unnecessary new convention.

May you advise what's the advantage of having?

Reply:
So, for me the `>>` and `<<` is much clearer to me. Bas raises a very valid point, that we should make a decision and make everything consistent (also the docs :-)

Reply:
The `<<` and `>>` operators were introduced in Airflow 1.8 and I've never really heard or seen anybody prefer using the "old" `set_upstream()` and `set_downstream()`. Especially when chaining multiple tasks your code looks the same as in the GUI and I think it's very pleasant to read:

```
task1 >> task2 >> task3 >> task4
```

Since my experience is people use the `>>` and `<<` operators much more, I figured it'd be useful to show that in the example DAGs, but correct me if wrong.

Reply:
Hi @Fokko @BasPH,

Ok, I'm quite surprised that these operators are already there since 1.8 while I have never seen it in the main tutorials/documentation.

I  think the main point to consider here is the `"education"`. Users like myself and @kaxil even didn't know the existence of these operators.

If we are going to do this, I would suggest to involve documentation "refactoring" as well. Make sure `>>` and `<<` operators are mentioned properly in the documentation/tutorials, so no users would be confused.

Reply:
There is a chapter on bitshift composition in the docs: [link here](https://airflow.apache.org/concepts.html?highlight=set_downstream#bitshift-composition).

However, the tutorial chapter indeed uses the `set_downstream` and `set_upstream` methods. I will check the documentation tonight and check where and how dependencies are set throughout the documentation.

Also, passing a list of tasks (e.g. `task1 >> [task2, task3]`) is something I learned from the example DAGs but I don't see it anywhere in the docs. Will update the docs on how dependencies can be set.

Reply:
👍

Reply:
I always wondered if :  task1 >> (task2, task3) can be supported.
Should be equivalent to:
task 1 >> task 2
task 1 >> task3

But I guess that is for a different topic.. just a thought

Reply:
@ron819 Setting dependencies with a tuple of tasks will also work because in `_set_relatives` the right-hand object is simply converted to a list ([code](https://github.com/apache/incubator-airflow/blob/39a9365b6250e588363e59dd6e332e02a28f411d/airflow/models.py#L3064)).

Reply:
@XD-DENG Well, I was unsure of the task list usage with rshift and lshift operators😂 not `>>` or `<<` (rshift or lshift) itself...

I think we already had good documentation over using them and I prefer using those as it resembles DAG + it is more clean. Many of our examples already do that.

Reply:
@kaxil I see I see. I still have so many more stuff to learn about Airflow ;-)

Reply:
> @ron819 Setting dependencies with a tuple of tasks will also work because in `_set_relatives` the right-hand object is simply converted to a list ([code](https://github.com/apache/incubator-airflow/blob/39a9365b6250e588363e59dd6e332e02a28f411d/airflow/models.py#L3064)).

Maybe this should be added to the docs as well as it isn't written anywhere.